drivers/hmis/spec/system/hmis/mci_clearance_spec.rb

```diff
@@ -0,0 +1,262 @@
+    fill_in 'Last Name', with: c1.last_name
+    mui_select 'Full name', from: 'Name Data Quality'
+    assert_text 'Not enough information to clear MCI'
+
+    # DOB
+    mui_date_select 'Date of Birth', date: 60.years.ago
+    mui_select 'Full DOB', from: 'DOB Data Quality'
+
+    # MCI component should show "MCI clearance available"
+    assert_text 'MCI Search Available'
+    # Scroll down to MCI section
+    element = find('button', text: 'Search for MCI ID')
+    execute_script('arguments[0].scrollIntoView(true)', element)
+  end
+
+  context 'Creating a new client globally' do
+    before(:each) do
+      allow(stub_mci).to receive(:clearance).and_return([])
+      visit '/client/new'
+    end
+
+    it 'creates new MCI ID by default' do
+      enter_client_details
+      click_button 'Search for MCI ID'
+      assert_text 'No Matches Found'
+      assert_text 'New MCI ID'
+      assert has_checked_field?('select_create_new_mci', disabled: true, visible: :all) # auto-selected
+
+      expect do
+        click_button 'Create Client'
+        assert_text c1.brief_name
+      end.to change(Hmis::Hud::Client, :count).by(1)
+
+      # Not mocking the actual ExternalId creation here, just ensuring that it was invoked
+      expect(stub_mci).to have_received(:create_mci_id)
+    end
+  end
+
+  context 'Editing an existing cleared client' do
+    before(:each) do
+      visit "/client/#{c1.id}/profile/edit"
+    end
+
+    it 'does not allow re-clearance' do
+      assert_text 'Client has been cleared'
+      assert_text c1_mci_id.value
+      assert_no_text 'Search for MCI ID'
+    end
+  end
+
+  context 'Editing an existing uncleared client' do
+    let!(:c2) { create :hmis_hud_client_complete, data_source: ds1, first_name: 'Jeremy', last_name: 'Goldwater' }
+    before(:each) do
+      visit "/client/#{c2.id}/profile/edit"
+    end
+
+    context 'with auto-clearance result for client that already exists in HMIS' do
+      before(:each) do
+        result = [
+          HmisExternalApis::AcHmis::MciClearanceResult.new(
+            mci_id: c1_mci_id.value,
+            score: 99, # above auto-clearance threshold (97)
+            client: build(:hmis_hud_client, first_name: 'quentin', man: 1),
+            existing_client_id: c1.id,
+          ),
+        ]
+        allow(stub_mci).to receive(:clearance).and_return(result)
+      end
+
+      it 'allows linking to MCI ID even though its a duplicate client' do
+        click_button 'Search for MCI ID'
+        assert_text 'Client already exists in HMIS'
+        assert has_field?("select_mci_#{c1_mci_id.value}", visible: :all, disabled: false) # dup is selectable
+        assert_no_text 'New MCI ID' # not allowed to create a new MCI ID, because they auto-cleared
+
+        # Choose the duplicate MCI ID (allow because editing, normally not allowed)
+        find_by_id("select_mci_#{c1_mci_id.value}", visible: :all).set(true)
+        expect do
+          click_button 'Save Changes'
```

Comment:
does this need to `assert` in this block so it waits until the action completes? (assert_text 'changes saved' or something)